src/main/java/net/snowflake/client/jdbc/SnowflakeUtil.java

```diff
@@ -813,4 +816,13 @@ public static boolean convertSystemPropertyToBooleanValue(
     }
     return defaultValue;
   }
+
+  @SnowflakeJdbcInternalApi
+  public static <T> T mapExceptions(ThrowingCallable<T, SFException> action) throws SQLException {
```

Comment:
Maybe this method name is too generic?

Reply:
Fixed

Reply:
I wanted to focus more on the specific exeptions :) I think about two solutions:
1. Making this function generic in case of exceptions mapped.
2. Renaming to something like `mapSFExceptionToSQLException`.
WDYT?

Reply:
1. I don't need now more generic.
2. Sounds good to contains exception types in name.